### DISCUSSION TITLE
error in check health core utils pwsh on windows

### DISCUSSION THREAD

**Discussion by Termanater13** [mason-org/mason.nvim#1599] [Q&A]
I'm on a Windows 11 PC running Neovim 0.9.5, with mason.nvim 1.9.0. When I run `:checkhealth` I get an error in Core utils:
```
- ERROR pwsh: not available
  - ADVICE:
    - spawn: pwsh failed with exit code 1 and signal 0. 
```
In the read me it says that pwsh or Powershell is required, and I do have PowerShell installed. I looked at the code tied to `:checkhealth` and to me it looks like it only checks for pwsh on windows. [mason.nvim/lua/mason/health.lua line:156](https://github.com/williamboman/mason.nvim/blob/e110bc3be1a7309617cecd77bfe4bf86ba1b8134/lua/mason/health.lua#L156) This is the start of the check if platform is windows. The only two checks I can see here is for pwsh and 7z.

I'm new to Neovim, sim still working out issues, at the moment this is my only error for mason.nvim, but I do have a bunch of warnings.

**Comment by StigInC** [comment]
Hey -- unsure if you solved this already, but I came across this post while having the same issue on Win10 and nvim11 (same `pwsh error` and `exit code 1 and signal 0` while trying to install cpptools DAP through `:Mason` -- with the `:MasonLog` saying something about `Import-Module Microsoft.PowerShell.Archive` failing) -- and I think I figured a solution from it (for myself at least). Maybe it'll help you or someone else. 

Reading the mason/health.lua link you posted (thank you, was unsure where to look for the Mason executions other than my kickstart-forked init.lua which didn't help), I noted both the check-and-command operation (`check { cmd = "pwsh",` you mentioned, as well as it running arguments such as `args = { --NoProfile`. So I came up with two solutions from this, one of which (or both) seemed to solve it. 

Even though I had a user environment variable to the folder already, I added an environment variable in PATH specifically to my pwsh executable (eg `C:\Users\...\PowerShell7\pwsh.exe`) since `"pwsh"` seems to be the exact name string it is looking for. **[Note for Windows 10 or below who do not already have it: `pwsh` seems to refer *specifically* to the new PowerShell7 and its `pwsh.exe` process; rather than the older PowerShell which comes default on older systems and has the process `powershell.exe`]** 

I also had the thought: `--"NoProfile",` might mean it's trying to rip a ~'Userless' instance of powershell to run the 7zip operation and on for finishing the install -- in which case it's possible my running nvim in Powershell was obstructing there being 'NoProfile', in addition to possible 'Microsoft Security' obstructions regarding extra-instance-spawning. So I closed Powershell and opened basic cmd (Command Prompt) instead, and ran nvim from there; and when I ran the cpptools DAP install from :Mason again from that cmd instance it worked without a problem. No elevations or anything needed.

I'm not sure which of these solved it, but it seems to have worked, so hopefully this helps you or anyone else who comes across this post searching for a solution to this problem.
